src/main/java/duke/data/TaskList.java

```diff
@@ -0,0 +1,76 @@
+package duke.data;
+
+import duke.commands.Task;
+import duke.exceptions.DukeException;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> lst;
+
+    public TaskList() {
+        lst = new ArrayList<Task>();
+    }
+
+    public void iterList() {
+        System.out.println("Here are the tasks in your list:");
+        if (lst.size() == 0) {
+            System.out.println("---- No Tasks currently ----");
+        }
+        int i = 1;
+        for (Task s : lst) {
+            System.out.println(i + "." + s);
+            i++;
+        }
+    }
+
+    public void add(Task text) {
+        lst.add(text);
+    }
+
+    public void complete(int index) throws DukeException {
+        if (index < 0 || index > lst.size() - 1) {
+            throw new DukeException("Hey! Invalid Task number given.");
+        }
+        Task task = lst.get(index);
+        task.completeTask();
+    }
+
+    public void delete(int index) throws DukeException {
+        if (index < 0 || index > lst.size() - 1) {
+            throw new DukeException("Hey! Invalid Task number given.");
+        }
+        lst.remove(index);
+    }
+
+    public ArrayList<Task> getList() {
+        return lst;
+    }
+
+    public int count() {
+        return lst.size();
+    }
+
+    public Task get(int index) throws DukeException {
+        if (index < 0 || index > lst.size() - 1) {
+            throw new DukeException("Hey! Invalid Task number given.");
+        }
+        Task task = lst.get(index);
+        return task;
+    }
+
+    public void searchKeyword(String keyword) {
```

Comment:
Should this be renamed to printMatchingTasks() so that it more accurately describes what the function does? Because the function actually prints the matching tasks to the screen.